internal/repository/skr/webhook/skrwebhook_repo.go

```diff
@@ -0,0 +1,166 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	skrwebhookresources "github.com/kyma-project/lifecycle-manager/internal/service/watcher/resources"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+)
+
+type SkrClientCache interface {
+	Get(key client.ObjectKey) client.Client
+}
+
+// ResourceRef contains the minimal information needed to identify and delete a resource.
+type ResourceRef struct {
+	Name       string
+	APIVersion string
+	Kind       string
+}
+
+type ResourceRepository struct {
+	resources           []ResourceRef
+	skrClientCache      SkrClientCache
+	remoteSyncNamespace string
+}
+
+func NewResourceRepository(
+	skrClientCache SkrClientCache,
+	remoteSyncNamespace string,
+	baseResources []*unstructured.Unstructured,
+) *ResourceRepository {
+	resources := make([]ResourceRef, 0, len(baseResources)+2)
+	for _, res := range baseResources {
+		gvk := res.GroupVersionKind()
+
+		resources = append(resources, ResourceRef{
+			Name:       res.GetName(),
+			APIVersion: gvk.GroupVersion().String(),
+			Kind:       gvk.Kind,
+		})
+	}
+
+	// Add generated resources that are created dynamically from SkrWebhookManifestManager (not read from resources.yaml)
+	resources = append(resources,
+		ResourceRef{
+			Name:       skrwebhookresources.SkrResourceName,
+			APIVersion: admissionregistrationv1.SchemeGroupVersion.String(),
+			Kind:       "ValidatingWebhookConfiguration",
+		},
+		ResourceRef{
+			Name:       skrwebhookresources.SkrTLSName,
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+	)
+
+	return &ResourceRepository{
+		resources:           resources,
+		skrClientCache:      skrClientCache,
+		remoteSyncNamespace: remoteSyncNamespace,
+	}
+}
+
+// ResourcesExist checks if any of the skr-webhook resources exist in the SKR cluster for the given Kyma.
+func (r *ResourceRepository) ResourcesExist(kymaName string) (bool, error) {
+	kymaNamespacedName := types.NamespacedName{
+		Name:      kymaName,
+		Namespace: r.remoteSyncNamespace,
+	}
+
+	skrClient := r.skrClientCache.Get(kymaNamespacedName)
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Add the existing context to the `ResourcesExist` func signature instaed of creating a new background context?

Reply:
I think we don't have a cancel from outside anyway. So would only be the ctx we receive from the Reconcile func, and we add the cancel within the func